pkg/sti/build.go

```diff
@@ -53,18 +57,19 @@ func newBuildHandler(req *Request) (*buildHandler, error) {
 		callbackInvoker: util.NewCallbackInvoker(),
 	}
 	rh.postExecutor = bh
+	rh.errorChecker = bh
 	return bh, nil
 }
 
-// Build processes a Request and returns a *Result and an error.
+// Build processes a Request and returns a *api.Result and an error.
 // An error represents a failure performing the build rather than a failure
 // of the build itself.  Callers should check the Success field of the result
 // to determine whether a build succeeded or not.
-func (b *Builder) Build() (*Result, error) {
+func (b *Builder) Build() (*api.Result, error) {
 	bh := b.handler
 	defer bh.cleanup()
 
-	err := bh.setup([]string{"assemble", "run"}, []string{"save-artifacts"})
+	err := bh.setup([]api.Script{api.Assemble, api.Run}, []api.Script{api.SaveArtifacts})
```

Comment:
The type is already called `Script` so I don't see any value in renaming those vars.